stl/inc/mutex

```diff
@@ -110,23 +93,16 @@ private:
     friend condition_variable;
     friend condition_variable_any;
 
-    union {
-        _Aligned_storage_t<_Mtx_internal_imp_size, _Mtx_internal_imp_alignment> _Mtx_storage;
-        _Mtx_internal_imp_mirror _Mtx_storage_mirror;
-    };
+    _Mtx_internal_imp_t _Mtx_storage{};
 
     _Mtx_t _Mymtx() noexcept { // get pointer to _Mtx_internal_imp_t inside _Mtx_storage

```

Comment:
Nitpick: This function comment has outlived its usefulness and is no longer particularly accurate - we should just remove it.